### DISCUSSION TITLE
MD simulation with NNP potential

### DISCUSSION THREAD

**Discussion by marzieh-saeedimasine** [MarsalekGroup/aml#17] [General]
I followed the from-trajectroy-to-model for my ZnO-water system and got weight values but when I applied for MD simulation with CP2K, it doesnt show stable dynamics. I wanted to know if its related to setting in CP2K input files. This is my input file and I am wondering if I need to introduce bias in METHOD  NNP part in force evaluation. Thanks to let me know your idea and if you can provide the setting for TiO2-water model, make it easy for me to compare. 

@SET PATH ../03-create-final-model/final-training/model
&GLOBAL
  PROJECT ZnO
  PRINT_LEVEL LOW
  RUN_TYPE MD
  WALLTIME 432000
&END GLOBAL

&MOTION
  &MD
    STEPS 50000
    TIMESTEP [fs] 0.5
    ENSEMBLE NVT
    &THERMOSTAT
      REGION MASSIVE
      TYPE NOSE
      &NOSE
        TIMECON [fs] 20
      &END NOSE
    &END THERMOSTAT
  &END MD
  &PRINT
    &TRAJECTORY
      &EACH
        MD 5
      &END EACH
    &END TRAJECTORY
    &VELOCITIES
      &EACH
        MD 5
      &END EACH
    &END VELOCITIES
    &FORCES
      &EACH
        MD 5
      &END EACH
    &END FORCES
    &RESTART OFF
    &END RESTART
    &RESTART_HISTORY OFF
    &END RESTART_HISTORY
  &END PRINT
&END MOTION

&FORCE_EVAL
  METHOD  NNP
  &NNP
    NNP_INPUT_FILE_NAME ${PATH}/nnp-000/input.nn
    SCALE_FILE_NAME  ${PATH}/nnp-000/scaling.data
    &MODEL
      WEIGHTS ${PATH}/nnp-000/weights
    &END MODEL
    &MODEL
      WEIGHTS ${PATH}/nnp-001/weights
    &END MODEL
    &MODEL
      WEIGHTS ${PATH}/nnp-002/weights
    &END MODEL
    &MODEL
      WEIGHTS ${PATH}/nnp-003/weights
    &END MODEL
    &MODEL
      WEIGHTS ${PATH}/nnp-004/weights
    &END MODEL
    &MODEL
      WEIGHTS ${PATH}/nnp-005/weights
    &END MODEL
    &MODEL
      WEIGHTS ${PATH}/nnp-006/weights
    &END MODEL
    &MODEL
      WEIGHTS ${PATH}/nnp-007/weights
    &END MODEL
    &PRINT
      &ENERGIES SILENT
        &EACH
          MD 1
        &END EACH
      &END ENERGIES
    &END PRINT
  &END NNP
  &SUBSYS
    &TOPOLOGY
      CONN_FILE_FORMAT OFF
      COORD_FILE_NAME out.xyz
      COORD_FILE_FORMAT XYZ
    &END TOPOLOGY
    &CELL
      A 10.619    0.000    0.000
      B  0.000   10.063    0.000
      C  0.000    0.000   36.180
    &END CELL
  &END SUBSYS
&END FORCE_EVAL

**Comment by cschran** [comment]
Your CP2K input file looks reasonable and for production runs you should not require energy biasing.

As outlined in my previous reply to your other discussion, your large force errors indicate that there could be a problem in your reference data, or in the reading of it by aml.

In this case I recommend taking a deeper look into the n2p2 output as well as correlation plots of the reference vs prediction. Please also confirm that the units of your reference data are as expected by aml (atomic units for the cp2k convenience reader).

**Reply by marzieh-saeedimasine** [reply]
Thanks for the answer. I used cp2k trajectory (in angstrom) and forces (in atomic unit, hartree/bohr) and cell (in atomic unit, bohr) for aml.read_frames_cp2k and aml.Structures.from_frames commands. When I checked input.data all values are in a.u.
It seems that input.data is good but as you mentioned the RMSE_Ftrain_pu is about 300 meV/A which is really high and its related to CP2K outputs. I realized that during AIMD, atoms are going out from the simulation box and I dont know if there is any tool for wrapping the atoms in aml when it reads the CP2K trajectory to return back them to the box. Does it cause any effect? please let me know your idea about it.

**Reply by OndrejMarsalek** [reply]
AML does not modify the positions it reads in any way, so no wrapping, either. That can be done either externally or as part of the Python code where one uses AML, if required. This is tangential, but CP2K normally does not care about atoms "outside the box" in periodic boundary conditions.

Generally, we can only reasonably address here issues related to AML itself, how the code behaves, possible bugs or feature requests, or similar topics. Broader research questions and issues related to software other than AML are best discussed in places intended for that (say, CP2K mailing list or n2p2 GitHub), where they also have the best chance to be resolved.

**Comment by OndrejMarsalek** [comment]
Just to emphasize the point that Christoph already made. The bias against disagreement that we proposed is really only intended for training set construction and model development. Once the model is complete it should not need biasing to remain stable.
